package/MDAnalysis/topology/guessers.py

```diff
@@ -354,6 +349,7 @@ def get_atom_mass(element):
     try:
         return tables.masses[element]
     except KeyError:
+        warnings.warn("Failed to guess the mass for the following atom types: {}".format(element))
```

Comment:
This will probably slow down topology parsing if one has many dummy atoms because the warning machinery is run for every single atom instead of a summary assessment in the original code.

For example, the TIP4P water model has one "MW" dummy atom in each molecule. One can have hundreds of thousands or millions of these, so it is important not to slow down the parsing any more than necessary.

Did you time the old and the new code on the example file with TIP4P (see below for the code – you can use `%timeit` or similar in ipython/jupyter for quick timing estimates):

```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import GRO

>>> u = mda.Universe(GRO)
~/anaconda3/envs/mda3/lib/python3.6/site-packages/MDAnalysis/topology/guessers.py:56: UserWarning: Failed to guess the mass for the following atom types: M

>>> u.atoms[u.atoms.masses == 0].names
array(['MW', 'MW', 'MW', ..., 'MW', 'MW', 'MW'], dtype=object)
```

As a side-note, reading from the equivalent TPR file (which contains all kind of additional information) does not raise the warning:
```python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import TPR, XTC

>>> u2 = mda.Universe(TPR, XTC)

>>> u2.atoms[u2.atoms.masses == 0].names
array(['MW', 'MW', 'MW', ..., 'MW', 'MW', 'MW'], dtype=object)
```

Reply:
I timed the existing implementation and the new implementation.

`Existing: 453 ms ± 9.49 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)`
`New: 476 ms ± 6.08 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)`

It is indeed slower.